sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVUtils.scala

```diff
@@ -93,6 +93,12 @@ object CSVUtils {
           value
         }
       }
+      if (header.sameElements(row)) {
+        header
+      } else {
+        // Ensure that the newly generated and existing headers are not duplicated.
+        makeSafeHeader(header, caseSensitive, options)
+      }
```

Comment:
Can you check how R's `read_csv` works in this case? That patch was inspired by R's one.

Reply:
Thanks for your review.

R uses`.` as the delimiter and a non-repeated increasing number as the suffix.
For example, the header is `a, a, a, a, a.2`
```R
> read.csv("x.csv", header = TRUE, sep = ",")
[1] a   a.1 a.3 a.4 a.2
```

Reply:
Can we follow this behaviour?

Reply:
Current behavior of Spark and R for CSV headers:

| CSV             | SPARK             | R               |
| --------------- | ----------------- | --------------- |
| `a,a,a,a`       | `a0,a1,a2,a3`     | `a,a.1,a.2,a.3` |
| `a,,,`          | `a,_c1,_c2,_c3`   | `a,X,X.1,X.2`   |
| *header: false* | `_c0,_c1,_c2,_c3` | `V1,V2,V3,V4`   |

If we follow R's behavior, we will introduce a user-facing change. This may cause errors in the user's legacy code. 
Maybe we should keep the behavior of Spark.

Reply:
I mean the numbering. Can. we create a name like `a1 a3 a4 a2` for `a, a, a, a, a.2`?

Reply:
Before that, spark added column index numbers as suffixes for duplicate column names, instead of adding incremental numbers as suffixes.
In my opinion, following R's behavior here may introduce an unnecessary user-facing change.
E.g:
`a0, b, c, a, d, a`
Follow the original behavior of Spark will be replaced with `a0, b, c, a3, d, a4`
The behavior that follows R will be replaced with `a0, b, c, a1, d, a2`